jobs/integr8ly/tower-clean-uninstall.yaml

```diff
@@ -0,0 +1,113 @@
+---
+
+- job:
+    name: tower-clean-uninstall
+    description: 'Uninstalls Integreatly from fresh pure cluster created by Ansible Tower tooling where Integreatly has never been installed.'
+    project-type: pipeline
+    sandbox: false
+    concurrent: true
+    properties:
+      - build-discarder:
+          num-to-keep: 20
+    parameters:
+      - string:
+          name: REPOSITORY
+          default: 'https://github.com/integr8ly/installation.git'
+          description: 'Repository of the Integreatly installer'
+      - string:
+          name: BRANCH
+          default: 'master'
+          description: 'Branch of the installer repository'
+      - string:
+          name: RECIPIENTS
+          default: integreatly-qe@redhat.com
+          description: 'Whitespace- or comma-separated list of recipient addresses'
+    triggers:
+      - timed: 'H H(0-3) * * *'
+    dsl: |
+
+        def setClusterAdminCredentials() {
+          def integreatlyCredentialsID = 'tower-openshift-cluster-credentials'
+          def clusterAdminCredentials = [:]
+          withCredentials([usernamePassword(credentialsId: integreatlyCredentialsID, usernameVariable: 'CLUSTER_ADMIN_USERNAME', passwordVariable: 'CLUSTER_ADMIN_PASSWORD')]) {
```

Comment:
How are `CLUSTER_ADMIN_USERNAME` and `CLUSTER_ADMIN_PASSWORD` provided to this pipeline?

Reply:
the `withCredentials` method creates and sets these two vars based on what is in the credentials so that you can use them in withCredentials block determined by curly brackets.

Reply:
I am wondering what would happen if the pipeline was executed on its own (there are no `CLUSTER_ADMIN_USERNAME` and `CLUSTER_ADMIN_PASSWORD` parameters)? or is it intended to be triggered by some other pipeline?

Reply:
The `CLUSTER_ADMIN_USERNAME` and `CLUSTER_ADMIN_PASSWORD` variables are created automatically by withCredentials method. And the values are populated with whatever is in credentials object identified by `integreatlyCredentialsID`. So the question is what would happen if there are no such credentials object set in Jenkins (no credentials having 'tower-openshift-cluster-credentials' as an ID). I don't know, so I am trying (build #17 ongoing atm). However, since the credentials with such ID are there, I don't think it matters that much.

Reply:
ok. Thanks for clarification

Reply:
This is the error if the credentials with such ID does not exist:
`org.jenkinsci.plugins.credentialsbinding.impl.CredentialNotFoundException: Could not find credentials entry with ID 'tower-openshift-cluster-credentials-DoesNotExist'`

Looks pretty nice, it is clear what is wrong.